### DISCUSSION TITLE
Support for Android Groovy Java

### DISCUSSION THREAD

**Discussion by Thazes** [patrykandpatrick/vico#901] [Questions]
### Question

I'm new to android I started learning using android java Groovy. Kindly clarify whether the library supports android java Groovy. Trying to modify the kotlin functions for setting the data entries is kinda confusing

### Vico version(s)

2.00 beta

### UI framework(s)

Views

**Comment by Gowsky** [comment]
Hello, @Thazes. Vico is a Kotlin library. It can be used with other JVM languages like Java and Groovy, since Kotlin automatically provides interoperability. However, it's not designed for other languages, so the experience won't be optimal. We recommend using Kotlin, which has been the standard language for Android for 7 years now.

**Comment by Thazes** [comment]
Hi thanks for the response. Can you please provide a sample implementation since I've got a delivery in a few days and I can't really convert my code into kotlin fully within the few days

**Reply by Gowsky** [reply]
You can check the sample app for the sample implementation using Kotlin. If you're asking about Groovy - we won't be able to provide you with any sample code.

**Reply by 4d7568616d6d6564** [reply]
The example I looked at was for Jetpack Compose, but I couldn't find a recommendation for Kotlin. @Gowsky

**Reply by Gowsky** [reply]
@4d7568616d6d6564, Jetpack Compose is still Kotlin. Do you mean the view system?
